google/stackdriver_monitoring/datadog_handlers.py

```diff
@@ -0,0 +1,137 @@
+
+    Args:
+      type_name: [string] The type of artifact.
+    """
+    super(ListArtifactsHandler, self).__init__(
+        url_path, command_name, description)
+    self.__type_name = type_name
+
+  def process_commandline_request(self, options):
+    """Implements CommandHandler."""
+    datadog = datadog_service.make_datadog_service(options)
+    if self.__type_name == 'screenboard':
+      method = datadog.api.Screenboard.get_all
+    elif self.__type_name == 'timeboard':
+      method = datadog.api.Timeboard.get_all
+    else:
+      raise ValueError('Unknown datadog artifact "{0}". '
+                       ' Either "screenboard" or "timeboard"'
+                       .format(self.__type_name))
+
+    json_text = json.JSONEncoder(indent=2).encode(method())
+    self.output(options, json_text)
+
+
+class GetArtifactHandler(BaseDatadogCommandHandler):
+  """Gets a specific datadog artifact instance .
+
+  Currently these are either Dashboards or Screenboards.
+  """
+  def __init__(self, type_name, url_path, command_name, description):
+    """Constructor.
+
+    Args:
+      type_name: [string] The type of artifact.
+    """
+    super(GetArtifactHandler, self).__init__(
+        url_path, command_name, description)
+    self.__type_name = type_name
+
+  def add_argparser(self, subparsers):
+    """Implements CommandHandler."""
+    parser = super(GetArtifactHandler, self).add_argparser(subparsers)
+    parser.add_argument(
+        '--name', required=True,
+      help='The name of the {0} to get.'.format(self.__type_name))
+    return parser
+
+  def process_commandline_request(self, options):
+    """Implements CommandHandler."""
+    title = options.get('name', None)
+    if not title:
+      raise ValueError('No name provided.')
+
+    api = datadog_service.make_datadog_service(options).api
+    if self.__type_name == 'screenboard':
+      list_method = api.Screenboard.get_all
+      get_method = api.Screenboard.get
+      result_key = 'screenboards'
+    elif self.__type_name == 'timeboard':
+      list_method = api.Timeboard.get_all
+      get_method = api.Timeboard.get
+      result_key = 'dashes'
+    else:
+      raise ValueError('Unknown datadog artifact "{0}". '
+                       ' Either "screenboard" or "timeboard"'
+                       .format(self.__type_name))
+
+    all_list = list_method()
+    artifact_id = None
+    for artifact in all_list[result_key]:
+      if artifact['title'] == title:
+        artifact_id = artifact['id']
+        break
+
+    if artifact_id is None:
+      raise ValueError('Could not find title "{0}"'.format(title))
+
+    json_text = json.JSONEncoder(indent=2).encode(get_method(artifact_id))
+    self.output(options, json_text)
+
+
+def add_handlers(handler_list, subparsers):
+  """Registers all the CommandHandlers for interacting with Datadog."""
+  command_handlers = [
+      ListArtifactsHandler('screenboard',
+                           None, 'list_datadog_screenboards',
```

Comment:
The command_name is the value of the 'command' option. It makes sense to call it command in the options. Should I call it command here? command_name seems to make more sense as an attribute name because it is a name (string) whereas a command sounds like a complex class.

The name is used in spinnaker_metric_tool.main to dispatch the command to the appropriate handler (and in command_processor.process_command to correlate the command against the handler instance to dispatch to).